api/comments/views.py

```diff
@@ -79,6 +79,7 @@ class CommentDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, Comm
         is_abuse       boolean            has this comment been reported by the current user?
         has_children   boolean            does this comment have replies?
         can_edit       boolean            can the current user edit this comment?
+        new_mentions   list               guids of mentioned users in content of comment
```

Comment:
>
I've been thinking through the idea of pushing the logic for extracting mentions from JavaScript down to Python. We'd still convert <span class="atwho"> to markdown before saving, but then we'd extract mentions from the markdown on the server side. The purpose would be to simplify the API: we wouldn't need *_mentions in the API at all.
>

Good idea! I think moving the `new_mentions` extraction to the model would make things much simpler and get rid of any `*_mentions` on the API. 
>
(maybe ever_mentioned would be a more descriptive name)
>

Changing the name to `ever_mentioned` would probably be less confusing.

>
That raises a bit of a question as to whether we would also push the conversion of `[@foo](/barry/)` into `<span class="atwho">` down into Python. I think asymmetry would make sense here, since atwho itself is totally a client-side concern.
>

I'm not sure where this would occur. `comment.js` loads the comments using the API and the API serializer sanitizes the content. Would it be after serialization, before the API returns data?